components/UploadPopover/UploadPopover.tsx

```diff
@@ -0,0 +1,208 @@
+import React, { useState } from "react";
+import styles from "./UploadPopover.module.css";
+import { Button, Popover, LinearProgress } from "@mui/material";
+import Icon from "../../assets/Icon";
+import { Transaction, User } from "../../types/schema";
+import { addTransaction } from "../../firebase/firestore/transaction";
+import Papa from '../../node_modules/papaparse';
+
+type UploadPopoverProps = {
+    uploadAnchor: Element;
+    closeUpload: Function;
+    popoverid: string;
+};
+
+export const UploadPopover: React.FunctionComponent<UploadPopoverProps> = ({
+  uploadAnchor,
+  closeUpload,
+  popoverid,
+}: UploadPopoverProps) => {
+    const [uploadFile, setUploadFile] = useState(null);
+    const [fileData, setFileData] = useState([]);
+    const [uploading, setUploading] = useState(false);
+    const [uploadProgress, setUploadProgress] = useState(0);
+    const [uploadSuccess, setUploadSuccess] = useState(false);
+
+    const sleep = (ms) => {
+        return new Promise(resolve => setTimeout(resolve, ms));
+    }
+
+    const handleUploadClose = () => {
+        closeUpload();
+        sleep(1000).then(() => {
+            setUploadSuccess(false);
+            setUploadFile(null);
+            setUploading(false);
+            setUploadProgress(0);
+        })
+    }
+
+    const handleUpload = (event) => {
```

Comment:
I tried clicking upload without adding any files, and it crashed :"( please look into this/perform error validation
<img width="1387" alt="Screen Shot 2021-11-16 at 12 38 31 PM" src="https://user-images.githubusercontent.com/28970991/142062036-a61f01db-182d-4395-ace9-0b31f3f6e510.png">